Add the `pipeline` subdocument to the `Sources` document (#8211)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/sources/pipeline.md

> ---

Reviewer: The `configuration` directory seems unnecessary. Just want to flag this for a future fix. This just makes the link longer (and shorter links are preferable for SEO). We can fix that by removing the `configuration` folder and adding redirects from the old link to the new link in all files under the `pipelines` directory. @vagimeli Could you please put in an issue to fix this?

Reviewer 2: Issue https://github.com/opensearch-project/documentation-website/issues/8223

layout: default
title: pipeline 
parent: Sources
grand_parent: Pipelines

> nav_order: 55

Reviewer: The topic names under "Sources" are not alphabetized. Could you fix that as part of this PR? Use Pascal numbering (multiples of 10).

Author: - I don't seem to see any directories following the so-called `Pascal numbering (multiples of 10)`. I made a new commit, should it look like this?

---

# pipeline

> Use the `pipeline` sink to write to another pipeline.

Reviewer: ```suggestion
Use the `pipeline` sink to read from another pipeline.
```

## Configuration options

The `pipeline` sink supports the following configuration options.

| Option | Required | Type   | Description                    |
|:-------|:---------|:-------|:-------------------------------|

> | name   | Yes      | String | Name of the pipe to read from. |

Reviewer: ```suggestion
| `name`   | Yes      | String | The name of the pipeline to read from. |
```

## Usage

> The following example configures a `pipeline` sink that reads from a pipeline named `movies`:

Reviewer: ```suggestion
The following example configures a `pipeline` sink named `sample-pipeline` that reads from a pipeline named `movies`:
```

> ```

Reviewer: ```suggestion
```yaml
```

sample-pipeline:
  source:
    - pipeline:
        name: "movies"

> ```

Reviewer: ````suggestion
```
{% include copy.html %}
````

Reviewer: Looking at the Data Prepper section, none of the yaml snippets have the `yaml` highlighter or copy button added. @vagimeli Could you please put in an issue to fix this?

Reviewer 2: Issue https://github.com/opensearch-project/documentation-website/issues/8223
